lib/model.js

```diff
@@ -2395,40 +2395,42 @@ class Model {
    *
    * **Implementation details:**
    *
-   * * MySQL - Implemented as a single query `INSERT values ON DUPLICATE KEY UPDATE values`
-   * * PostgreSQL - Implemented as a temporary function with exception handling: INSERT EXCEPTION WHEN unique_constraint UPDATE
-   * * SQLite - Implemented as two queries `INSERT; UPDATE`. This means that the update is executed regardless of whether the row already existed or not
+   * * MySQL - Implemented with ON DUPLICATE KEY UPDATE`
+   * * PostgreSQL - Implemented with ON CONFLICT DO UPDATE. If update data contains PK field, then PK is selected as the default conflict key. Otherwise first unique constraint/index will be selected, which can satisfy conflict key requirements.
+   * * SQLite - Implemented with ON CONFLICT DO UPDATE
    * * MSSQL - Implemented as a single query using `MERGE` and `WHEN (NOT) MATCHED THEN`
-   * **Note** that SQLite returns undefined for created, no matter if the row was created or updated. This is because SQLite always runs INSERT OR IGNORE + UPDATE, in a single query, so there is no way to know whether the row was inserted or not.
+   *
+   * **Note** that Postgres/SQLite returns null for created, no matter if the row was created or updated
    *
    * @param  {object}       values                                        hash of values to upsert
    * @param  {object}       [options]                                     upsert options
    * @param  {boolean}      [options.validate=true]                       Run validations before the row is inserted
    * @param  {Array}        [options.fields=Object.keys(this.attributes)] The fields to insert / update. Defaults to all changed fields
    * @param  {boolean}      [options.hooks=true]                          Run before / after upsert hooks?
-   * @param  {boolean}      [options.returning=false]                     If true, fetches back auto generated values (Postgres only)
+   * @param  {boolean}      [options.returning=true]                      If true, fetches back auto generated values
```

Comment:
Does this work for all 5 dialects now?

Reply:
Except for mariadb, which does not support `RETURINING *` at all, in current 10.1 target version. But it wont fail

Reply:
I am confused, if it is not supported by MariaDB, then why could the following code be removed:

```js
    if (options.returning === true && primaryKey) {
      const record = await this.findByPk(primaryKey, options);
      result = [record, created];
```

Reply:
This works but it is buggy, see linked issues. Plus one extra read from database. So I decided to not support this for mariadb.